packages/scandipwa/src/store/Wishlist/Wishlist.dispatcher.js

```diff
@@ -160,43 +160,46 @@ export class WishlistDispatcher {
             );
     }
 
-    moveWishlistToCart(dispatch, sharingCode) {
+    async moveWishlistToCart(dispatch, sharingCode) {
         if (!isSignedIn()) {
             return Promise.reject();
         }
 
-        return fetchMutation(WishlistQuery.getMoveWishlistToCart(sharingCode))
-            .then(
-                /** @namespace Store/Wishlist/Dispatcher/WishlistDispatcher/moveWishlistToCart/fetchMutation/then */
-                () => {
-                    this._syncWishlistWithBE(dispatch);
-                    CartDispatcher.then(
-                        ({ default: dispatcher }) => dispatcher.updateInitialCartData(dispatch)
-                    );
-                }
+        try {
+            return await fetchMutation(WishlistQuery.getMoveWishlistToCart(sharingCode));
+        } finally {
+            await this._syncWishlistWithBE(dispatch);
+            CartDispatcher.then(
+                ({ default: dispatcher }) => dispatcher.updateInitialCartData(dispatch)
             );
+            dispatch(showNotification('success', __('Available items moved to cart')));
+        }
     }
```

Comment:
1. Await in the try block is redudant here. You already return a promise.
2. Did you check the finally block here? I believe it will fire before `fetchMutation` will have a response.